### PR TITLE
functional whisper chat

### PR DESCRIPTION
**Pull request by gluk256**
In order to actually use the chat, you need to change the ALeth.cpp file, ln.107: uncomment the "shh" param.

### CODE REVIEW

File: alethzero/ZeroFace.h [hunk 1]
```diff
@@ -71,6 +71,7 @@ class ZeroFace: public QMainWindow
 	void killPlugins();
 	void noteAllChange();
 	virtual void addSettingsPage(int _index, QString const& _categoryName, std::function<SettingsPage*()> const& _pageFactory) = 0;
+	std::shared_ptr<Plugin> findPlugin(std::string const& _name) { auto it = m_plugins.find(_name); return it != m_plugins.end() ? it->second : 0; }
```
Comment (chfast) [inline alethzero/ZeroFace.h]: Replace `0` with `nullptr`.

**Comment by gluk256** [comment]
fixed

File: alethzero/plugins/whisper/WhisperPeers.cpp [hunk 2]
```diff
@@ -37,6 +38,32 @@ using namespace zero;
...
+static QString const c_filterDecrypted("-= all I can decrypt =-");
+static QString const c_filterToMe("-= messages addressed to me =-");
+static QString const c_filterAll("-= show all messages =-");
+
+enum { SpecificTopic = 1, KnownTopics = 2, AddressedToMe = 4, AllMessages = 8 };
+
+QString messageToString(shh::Envelope const& _e, shh::Message const& _m)
+{
+	time_t birth = _e.expiry() - _e.ttl();
+	QString t(ctime(&birth));
```
Comment (arkpar) [inline alethzero/plugins/whisper/WhisperPeers.cpp:50]: Use locale enabled function for date/time formatting (strftime)

**Comment by gluk256** [comment]
This is just the first iteration. The chat format is subject to change. I believe, fine-tuning is premature at this stage.

**Comment by LianaHus** [comment]
looksgood

**Pull request merged**